eth2util/eth2exp/attagg_old.go

```diff
@@ -44,8 +44,44 @@ func IsAggregator(ctx context.Context, specProvider eth2client.SpecProvider, com
 		modulo = 1
 	}
 
+	return hash(slotSig, modulo)
+}
+
+// IsSyncCommAggregator returns true if the validator is the aggregator for the provided sync subcommittee.
+// Refer: https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/validator.md#aggregation-selection
+func IsSyncCommAggregator(ctx context.Context, specProvider eth2client.SpecProvider, sig eth2p0.BLSSignature) (bool, error) {
+	spec, err := specProvider.Spec(ctx)
+	if err != nil {
+		return false, errors.Wrap(err, "get eth2 spec")
+	}
+
+	commSize, ok := spec["SYNC_COMMITTEE_SIZE"].(uint64)
+	if !ok {
+		return false, errors.New("invalid SYNC_COMMITTEE_SIZE")
+	}
+
+	commSubnetCount, ok := spec["SYNC_COMMITTEE_SUBNET_COUNT"].(uint64)
+	if !ok {
+		return false, errors.New("invalid SYNC_COMMITTEE_SUBNET_COUNT")
+	}
+
+	aggsPerComm, ok := spec["TARGET_AGGREGATORS_PER_SYNC_SUBCOMMITTEE"].(uint64)
+	if !ok {
+		return false, errors.New("invalid TARGET_AGGREGATORS_PER_SYNC_SUBCOMMITTEE")
+	}
+
+	modulo := commSize / commSubnetCount / aggsPerComm
+	if modulo < 1 {
+		modulo = 1
+	}
+
+	return hash(sig, modulo)
+}
+
+// hash returns true if the first 8 bytes of the sha256 hash of the input signature is divisible by the provided modulo.
+func hash(sig eth2p0.BLSSignature, modulo uint64) (bool, error) {
```

Comment:
hash is not a good name, maybe `sigSelected` or `moduloSelect` or `hashModulo`

Reply:
cool, going with `hashModulo` as it better explains that this function performs operations like hashing and calculating modulo